service/kas/access/rewrap.go

```diff
@@ -345,19 +345,60 @@ func (p *Provider) tdf3Rewrap(ctx context.Context, body *RequestBody, entity *en
 	}, nil
 }
 
-func (p *Provider) nanoTDFRewrap(body *RequestBody) (*kaspb.RewrapResponse, error) {
+func (p *Provider) nanoTDFRewrap(ctx context.Context, body *RequestBody, entity *entityInfo) (*kaspb.RewrapResponse, error) {
 	headerReader := bytes.NewReader(body.KeyAccess.Header)
 
-	header, err := sdk.ReadNanoTDFHeader(headerReader)
+	header, _, err := sdk.NewNanoTDFHeaderFromReader(headerReader)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse NanoTDF header: %w", err)
 	}
 
-	symmetricKey, err := p.CryptoProvider.GenerateNanoTDFSymmetricKey(header.EphemeralPublicKey.Key)
+	symmetricKey, err := p.CryptoProvider.GenerateNanoTDFSymmetricKey(header.EphemeralKey)
 	if err != nil {
 		return nil, fmt.Errorf("failed to generate symmetric key: %w", err)
 	}
 
+	// check the policy binding
+	digest := ocrypto.CalculateSHA256(header.EncryptedPolicyBody)
+	binding := digest[len(digest)-kNanoTDFGMACLength:]
+	if !bytes.Equal(binding, header.PolicyBinding) {
+		return nil, fmt.Errorf("policy binding check failed")
+	}
```

Comment:
Don't we need the actual policy to check the binding or is it different in nanotdf?

Reply:
The actual policy is checked starting on the next line